### ISSUE TITLE
Error: [$compile:ctreq] Controller 'mdDataTable', required by directive 'mdTableRow', can't be found!

### ISSUE THREAD

**Issue by hannupekka** [daniel-nagy/md-data-table#152]
I'm getting this error with Safari 9 while newest Chrome works fine. Below is the template for my table.
```
<md-data-table-container class="md-whiteframe-z1">
  <table md-data-table>
    <thead>
      <tr>
        <th name="Group"></th>
        <th name="Name"></th>
        <th name="Description"></th>
        <th name="Created at"></th>
      </tr>
    </thead>
    <tbody>
      <tr ng-repeat="form in searchForms()">
        <td>{{ form.group }}</td>
        <td>{{ form.name }}</td>
        <td>{{ form.comment }}</td>
        <td>{{ form.created }}</td>
      </tr>
    </tbody>
  </table>
</md-data-table-container>
```

Any idea what is causing this? Using Browserify but I guess everything is required correctly since Chrome works.

**Comment by ebrahim-tadbir** [comment]
+1
I have same problem, I'm using md-data-table in md-tab, when I switch tabs I have this problem.

**Comment by grogone** [comment]
I had this problem in Chrome after code minification. I've noticed that in the view an element before `md-data-table-container` had wrong closing tag.

```
  <h2 class="md-title" translate="SHOP_SELECT_LP_TO_ADD"></h1>
```

So I fixed it.

 ```
  <h2 class="md-title" translate="SHOP_SELECT_LP_TO_ADD"></h2>
```

And it's work for me. I didn't dig deeper. May it help.

**Comment by PauloLira** [comment]
+1

**Comment by pdore-netfore** [comment]
+1

**Comment by Splaktar** [comment]
I keep getting the following in Chrome stable:

`Controller 'mdTable', required by directive 'mdCell', can't be found!`

Still debugging in... seems to be an issue with md-data-table or our use of it though.

**Comment by daniel-nagy** [comment]
@Splaktar are you using angular `>=1.4.0`. Also, if you are using ui-router this is a known issue #325.

**Comment by Splaktar** [comment]
Yes, using Angular 1.4.8 or 1.5.x. Not using ui-router on this project just angular-route.

**Comment by LeonVay** [comment]
Hello everyone,
I have faced the same problem: Controller 'mdDataTable', required by directive 'mdTableRow', can't be found!
On different versions of angular (1.5.X) there are different messages, but the root cause seems to be the same.
Switching off animation does not helped.

**Comment by twizzzlers** [comment]
Happening for me too +1

Controller 'mdTable', required by directive 'mdCell', can't be found!

**Comment by RooTooZ** [comment]
:+1:

**Comment by elekzalan** [comment]
+1

**Comment by tvm18860** [comment]
+1 also using ui-router

**Comment by paulhauner** [comment]
I'm also experiencing the issue. Similar to @ebrahim-tadbir I am getting the issue when switching md-tabs.

**Comment by Maldris** [comment]
Has anyone had any luck overcoming this issue, or making a work around? I've been pulled onto updating an old project and we need some recent features of angular-material, which forces us to update angular beyond where md-data-table breaks.
looking at angular 1.5.8 with ngRouter not ui-router.
I have tried @mikedon's workaround from the other thread @daniel-nagy linked to above, at least in my case it wasn't appropriate. while it did remove the error, md-data-table stopped working.
I also have an old minified md-data-table.min.js file of unknown version from the old server, while it doesn't throw the error, it still doesn't render correctly under the newer angular version.
does anyone have any recommendations?

**Comment by Maldris** [comment]
I found that the reason the original version wasn't working was that I still had the newer css being pulled, reverting that made it draw fine for now, ideally I would like a suggested fix for this problem anyway so updates can be kept consistent.

**Comment by DominicGaribaldi** [comment]
+1

**Comment by jpca999** [comment]
+1  same issue.

**Comment by mattbajorek** [comment]
+1 only when webpack is building

**Comment by Kerimr** [comment]
Check this. Maybe it helps
[](https://github.com/daniel-nagy/md-data-table/issues/623#issuecomment-482099442)
